assets/dns/daemonset.yaml

```diff
@@ -44,14 +44,56 @@ spec:
           timeoutSeconds: 5
           successThreshold: 1
           failureThreshold: 5
+      - name: dns-node-resolver
+        # image is set at runtime
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: hosts-file
+          mountPath: /etc/hosts
+        # env NAMESERVER and CLUSTER_DOMAIN are set at runtime
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -euo pipefail
+
+          REGISTRY_SERVICE="docker-registry.default.svc"
+          OPENSHIFT_MARKER="openshift-generated-node-resolver"
+          HOSTS_FILE="/etc/hosts"
+          TEMP_FILE="/etc/hosts.tmp"
+
+          while true; do
+            # Fetch registry service IP from cluster dns if present
+            ips=($(dig @"${NAMESERVER}" +short "${REGISTRY_SERVICE}.${CLUSTER_DOMAIN}"))
+            if [[ ("$?" -eq 0) && ("${#ips[@]}" -ne 0) ]]; then
```

Comment:
Design question: If the name does not resolve to anything (i.e., `[[ "${#ips[@]" -eq 0 ]]`), should we still regenerate `/etc/hosts` in order to clear out any stale entries? Or do we want to keep stale entries in order to avoid turning an API or CoreDNS outage into a registry outage? Whatever decision we make on the design, I suggest we add a comment here to document it.